### PR TITLE
Generate junit reports for ginkgo tests

### PR DESCRIPTION
**Pull request by mguetta1**
Ginkgo supports generating JUnit reports with `ginkgo --junit-report=report.xml`
To make the XML report files more clear and avoid duplicates of test runs - I changed the execution of the different feature tests to run as standalone ginkgo suite, by running for example:
```cd e2e/jiraintegration/ && ginkgo -v --junit-report=jiraintegration-report.xml --output-dir=${GINKGO_REPORT_DIR}```

This approach forced the removal of configuration mechanism by config.json file due to that ginkgo command enters to the suite folder and test fails to find the config file. So only env vars are supported now

cc: @msajidmansoori12

### CODE REVIEW

**Comment by mguetta1** [comment]
Jenkins run for example:
https://main-jenkins-csb-migrationqe.apps.ocp-c1.prod.psi.redhat.com/view/MTA/job/mta/job/konveyor-go-api-tests-runner/12/console

Next is to merge the XML report files and display a clear output of passed/failed/skipped tests (by DevOps QE)

File: Makefile [hunk 1]
```diff
@@ -58,18 +58,15 @@ test-analysis:
...
-	ginkgo -v ./e2e/metrics/...
+	cd e2e/metrics/ && ginkgo -v --junit-report=metrics-report.xml --output-dir=${GINKGO_REPORT_DIR}
 
 # Jira Integration.
 test-jira:
-	@if [ -z "${JIRA_CLOUD_USERNAME}" ]; then echo "Error: JIRA_CLOUD_USERNAME environment variable is not defined."; exit 1; fi
-	@if [ -z "${JIRA_CLOUD_PASSWORD}" ]; then echo "Error: JIRA_CLOUD_PASSWORD environment variable is not defined."; exit 1; fi
-	@if [ -z "${JIRA_CLOUD_URL}" ]; then echo "Error: JIRA_CLOUD_URL environment variable is not defined."; exit 1; fi
-	ginkgo -v -focus "Jira cloud"
+	cd e2e/jiraintegration/ && ginkgo -v --junit-report=jiraintegration-report.xml --output-dir=${GINKGO_REPORT_DIR}
```
Comment (aufi) [inline Makefile:65]: Just curious, isn't `-focus` be missing there?

File: Makefile [hunk 2]
```diff
@@ -44,7 +45,6 @@ test-tier2:
 
 # TIER3
 test-tier3:
-	TIER3=1 MAVEN_TESTAPP_USER=$(GITHUB_USER) MAVEN_TESTAPP_TOKEN=$(GITHUB_TOKEN) $(MAKE) test-analysis
```
Comment (aufi) [inline Makefile:47]: Is the TIER3 analysis test going ot be executed in other target or we're getting rid of analysis test on non-public environment?

File: Makefile (hunk 1 above)
Comment (mguetta1) [reply Makefile:65]: Hi, no.
`-focus "<description>"` is used to run any Ginkgo spec (test) that contains the "description" as the spec's description (or part of)
Once there is a suite file (`jiraintegration_suite_test.go`) in the directory and ginkgo command is executed inside the directory or includes the directory path - any ginkgo spec within the directory will be run (unless you provide `-focus` to run a specific spec within the current suite)

File: Makefile (hunk 2 above)
Comment (mguetta1) [reply Makefile:47]: TIER3 is being executed on Jenkins as PR check and there is an open task to run it as nightly as well against the `main` branch

Comment (aufi) [reply Makefile:47]: Ok, thanks!

**Review by aufi — APPROVED** [review]
LGTM

**Pull request merged**